src/Identity/Extensions.Core/src/UserManager.cs

```diff
@@ -2276,7 +2277,59 @@ public virtual string GenerateNewAuthenticatorKey()
     /// </summary>
     /// <returns></returns>
     protected virtual string CreateTwoFactorRecoveryCode()
-        => Guid.NewGuid().ToString().Substring(0, 8);
+    {
+        var recoveryCode = new StringBuilder(11);
```

Comment:
Since it's a fixed-length string, maybe we could use `char[]` instead of `StringBuilder`?

Reply:
Since the goal is to generate a string, in that case it wouldn't matter much, SB is allocating an array internally and then generates a single string out of it.